src/Api/DataProvider/TpayChannelItemDataProvider.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CommerceWeavers\SyliusTpayPlugin\Api\DataProvider;
+
+use ApiPlatform\Core\DataProvider\ItemDataProviderInterface;
+use ApiPlatform\Core\DataProvider\RestrictedDataProviderInterface;
+use CommerceWeavers\SyliusTpayPlugin\Api\Resource\TpayChannel;
+use CommerceWeavers\SyliusTpayPlugin\Tpay\Provider\TpayApiBankListProviderInterface;
+
+final class TpayChannelItemDataProvider implements ItemDataProviderInterface, RestrictedDataProviderInterface
+{
+    public function __construct(
+        private readonly TpayApiBankListProviderInterface $apiBankListProvider,
+    ) {
+    }
+
+    public function getItem(string $resourceClass, $id, string $operationName = null, array $context = []): ?TpayChannel
+    {
+        $transactionChannels = $this->apiBankListProvider->provide();
+
+        /** @var array $transactionChannel */
+        foreach ($transactionChannels as $transactionChannel) {
+            if ($transactionChannel['id'] === $id) {
+                return TpayChannel::fromArray($transactionChannel);
+            }
+        }
```

Comment:
It'd be cool if the provided list will be string-indexed and the index will be the identifier, so you can just use `array_key_exists` and `[$id]`.